syntax/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+#![no_std]
+
+#[macro_use]
```

Comment:
nit: Do we actually need this?

Reply:
AFAIK yes, in order to import the `macro_rules!` macros from `liballoc`. Normally `rustc` will complain if `#[macro_use]` is unnecessary, but it is not doing so here, so I assume it is needed, but I haven't tried removing it.

I'll see what happens if I do, and if it doesn't break anything, I'll strip it out. It costs nothing to have it in any case (it simply brings the macros into scope, it doesn't imply any extra code linked in to the actual library).